signer/tests/integration/postgres.rs

```diff
@@ -5580,3 +5580,1200 @@ async fn is_withdrawal_inflight_catches_withdrawals_in_package() {
+            ..Faker.fake()
+        })
+        .await
+        .expect("failed to write withdrawal reject event");
+    }
+
+    /// Creates a sweep transaction that includes the specified withdrawal
+    /// request. The sweep transaction is written to the database in the
+    /// provided bitcoin block and the transaction ID is returned.
+    async fn sweep_withdrawal_request(
+        db: &PgStore,
+        request: &WithdrawalRequest,
+        at_bitcoin_block: &BitcoinBlockHash,
+    ) -> BitcoinTxId {
+        // Simulate a sweep transaction on the canonical chain which will
+        // include the withdrawal request.
+        let transaction = model::Transaction {
+            txid: Faker.fake(),
+            block_hash: at_bitcoin_block.into_bytes(),
+            tx_type: model::TransactionType::SbtcTransaction,
+            tx: Vec::new(),
+        };
+        let bitcoin_sweep_tx = model::BitcoinTxRef {
+            txid: transaction.txid.into(),
+            block_hash: *at_bitcoin_block,
+        };
+        db.write_transaction(&transaction)
+            .await
+            .expect("failed to write transaction");
+        db.write_bitcoin_transaction(&bitcoin_sweep_tx)
+            .await
+            .expect("failed to write bitcoin transaction");
+
+        // Write a fully validated withdrawal output for the request.
+        db.write_bitcoin_withdrawals_outputs(&[model::BitcoinWithdrawalOutput {
+            bitcoin_txid: bitcoin_sweep_tx.txid,
+            bitcoin_chain_tip: *at_bitcoin_block,
+            is_valid_tx: true,
+            stacks_txid: request.txid,
+            stacks_block_hash: request.block_hash,
+            request_id: request.request_id,
+            validation_result: WithdrawalValidationResult::Ok,
+            output_index: 2,
+        }])
+        .await
+        .expect("failed to write bitcoin withdrawal output");
+
+        transaction.txid.into()
+    }
+
+    /// Asserts that when there are no withdrawal requests, we return an empty
+    /// list (just to make sure we don't error anywhere).
+    #[test_log::test(tokio::test)]
+    async fn returns_empty_list_when_no_requests() {
+        let db = signer::testing::storage::new_test_database().await;
+
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(&Faker.fake(), &Faker.fake(), 1_000, 0)
+            .await
+            .expect("failed to query db");
+        assert!(requests.is_empty());
+
+        signer::testing::storage::drop_db(db).await;
+    }
+
+    /// Asserts that requests which are confirmed in stacks blocks whose anchor
+    /// block is, while in the canonical bitcoin chain, too old to be considered
+    /// accepted and are not returned.
+    ///
+    /// The signature threshold/context window we are testing is _inclusive_.
+    ///
+    /// This test creates blockchains with the following structure:
+    ///
+    /// ```text
+    /// Age:         3           2           1     
```

Comment:
nit: age it's confusing since we deal with heights below

Reply:
Oh, right, leftover from the previous version

Reply:
[eceeb7f](https://github.com/stacks-network/sbtc/pull/1397/commits/eceeb7f6db1f6392f9363e0135f883301a7241d4)